File: minimum-size-subarray-sum/minimum-size-subarray-sum.rb
# @param {Integer} s
# @param {Integer[]} nums
# @return {Integer}
def min_sub_array_len(s, nums)
    
    i, j = 0, 0
    sum, minLength = 0, nums.length+1
    
    while j < nums.length do
        sum += nums[j]
        j+=1
        
        while sum >= s do
            minLength = [minLength, j-i].min
            sum -= nums[i]
            i += 1
        end
    end 
        
    return minLength == nums.length+1 ? 0: minLength
    
end

File: minimum-size-subarray-sum/minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int maxLen = Integer.MAX_VALUE;
        
        for (int i = 0; i < nums.length; i++) {
            int sum = 0;
            for (int j = i; j < nums.length; j++) {
                sum += nums[j];
                if (sum >= target) {
                    maxLen = Math.min(maxLen, j-i+1);
                    break;
                }
            }
            
        }
        
        
                
        return maxLen == Integer.MAX_VALUE ? 0 : maxLen;
    }
}